arrows/ceres/register_algorithms.cxx

```diff
@@ -33,26 +33,39 @@
  * \brief Ceres algorithm registration implementation
  */
 
-#include "register_algorithms.h"
+#include <arrows/ceres/kwiver_algo_ceres_export.h>
+#include <vital/algo/algorithm_factory.h>
 
-
-#include <arrows/algorithm_plugin_interface_macros.h>
 #include <arrows/ceres/bundle_adjust.h>
 
 
 namespace kwiver {
 namespace arrows {
 namespace ceres {
 
-/// Register Ceres algorithm implementations with the given or global registrar
-int register_algorithms( vital::registrar &reg )
+extern "C"
+KWIVER_ALGO_CERES_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  REGISTRATION_INIT( reg );
+  static auto const module_name = std::string( "algorithms.ceres" );
+  if (vpm.is_module_loaded( module_name ) )
```

Comment:
I'm not sure if this is a problem or not, but I think this behavior might be different than the old implementation.  I think with the old plugin loader a module with the same name as a previously loaded module would replace the previously loaded module.  This looks like a module with the same name will be skipped, keeping the old modules.

Again, I'm not sure how important this issue is, but it could come into play when you have multiple versions of the same plugin in different paths.  E.g. maybe I have an installed Ceres plugin, but I also have a modified version in my build directory which I've added to my KWIVER_PLUGIN_PATH.